solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -100,6 +100,44 @@ pub mod audius_data {
         Ok(())
     }
 
+    /// Functionality to create user without admin privileges
+    pub fn create_user(ctx: Context<CreateUser>, user_authority: PubKey) -> ProgramResult {
+        msg!("Audius::CreateUser");
+
+        // Confirm that the base used for user account seed is derived from this Audius admin storage account
+        let (derived_base, _) = find_program_address_pubkey(ctx.accounts.admin.key(), ctx.program_id);
+        if derived_base != base {
+            return Err(ErrorCode::Unauthorized.into());
+        }
+
+        // Confirm that the derived pda from base is the same as the user storage account
+        let (derived_user_acct, _) = Pubkey::find_program_address(&[&derived_base.to_bytes()[..32], &handle_seed], ctx.program_id);
```

Comment:
instead of doing this here, let's leverage the `seeds` trait in the instruction definition and also verify the bump_seed - https://github.com/AudiusProject/audius-protocol/blob/master/solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs#L300

Reply:
if you don't mind removing this from `init_user` that would be great as well, but that already has trait verification in addition to manually verifying the output of `find_program_address`